src/ui/SmartSnippet/ExplanationModal.ts

```diff
@@ -0,0 +1,109 @@
+import { AccessibleModal } from '../../utils/AccessibleModal';
+import { l } from '../../strings/Strings';
+import { $$ } from '../../utils/Dom';
+import { RadioButton } from '../FormWidgets/RadioButton';
+import 'styling/_ExplanationModal';
+
+export interface IExplanation {
+  label: string;
+  onSelect: () => void;
+  hasDetails?: boolean;
+}
+
+export interface IExplanationModalOptions {
+  ownerElement: HTMLElement;
+  explanations: IExplanation[];
+  onClosed: () => void;
+  modalBoxModule?: Coveo.ModalBox.ModalBox;
+}
+
+const ROOT_CLASSNAME = 'coveo-user-explanation-modal';
+const CONTENT_CLASSNAME = `${ROOT_CLASSNAME}-content`;
+const EXPLANATIONS_CLASSNAME = `${ROOT_CLASSNAME}-explanations`;
+const DETAILS_SECTION_CLASSNAME = `${ROOT_CLASSNAME}-details`;
+const DETAILS_TEXTAREA_CLASSNAME = `${DETAILS_SECTION_CLASSNAME}-textarea`;
+const DETAILS_LABEL_CLASSNAME = `${DETAILS_SECTION_CLASSNAME}-label`;
+const SEND_BUTTON_CLASSNAME = `${ROOT_CLASSNAME}-send-button`;
+
+export class ExplanationModal {
+  private modal: AccessibleModal;
+  private explanationRadioButtons: RadioButton[];
+  private selectedExplanation: IExplanation;
```

Comment:
I think `reasons` instead of `explanationRadioButtons`, and `selectedReason` instead of `selectedExplanation` are more inline with the wording in other parts of the PR.